### PR TITLE
Add NN paging and show/hide non default nets

### PR DESCRIPTION
**Pull request by tomtor**
@vondele As a followup to #816 some quick paging. Not sure if we really need this, `/users` is more than 1000 entries and works fine too, but better safe than sorry.

@ppigazzini I did some creative testing of this. You could lower page_size by hand on dev to 100 for testing....

### CODE REVIEW

**Comment by tomtor** [comment]
@vondele We could lower the `500` pagesize somewhat now that we have paging?

**Comment by ppigazzini** [comment]
@tomtor DEV updated with `page_size = 5` for hallucinated testing, some pages have lesser than 5 entries (???).

**Comment by tomtor** [comment]
> some pages have lesser than 5 entries

Some entries are hidden when not logged in

**Comment by tomtor** [comment]
@ppigazzini Added show/hide of non default nets...

**Comment by tomtor** [comment]
@ppigazzini Forgot to bump the javascript version....

**Comment by ppigazzini** [comment]
DEV updated with 50 pages. LGTM.

**Comment by tomtor** [comment]
With 50 i get only one green network one the first page when hiding non-default nets.

**Comment by ppigazzini** [comment]
> With 50 i get only one green network one the first page when hiding non-default nets.

I thought that intended action was to hide the non default nets of the page.

**Comment by tomtor** [comment]
> I thought that intended action was to hide the non default nets of the page.

Sure, I meant that lowering from 250 to 50 is not a good idea.

I will squash the PR, I think it's OK to merge.

**Comment by ppigazzini** [comment]
50 is creative testing :)

**Comment by vondele** [comment]
I'm maybe a bit late with the comment, what about using 'earlier networks' instead of 'more networks' and I'm wondering if we should have a way to go back 'later networks'. 
I would have a slight preference for showing all nets by default.

**Comment by tomtor** [comment]
@ppigazzini Added Joost's suggestions and changed page size to 250

**Comment by ppigazzini** [comment]
DEV updated, page=50 for creative debug, I will swap the "earlier" and "later" positions to have the same order of pages from pagination().

BTW I applied unsuccessfully my js coding skillset (copy&paste&hope_it_works) with the pagination code :)

https://github.com/glinscott/fishtest/blob/c7acb76feed19f3cf96bed5f27e7e1c15d02799c/fishtest/fishtest/templates/run_table.mak#L39-L59

**Comment by tomtor** [comment]
> I applied unsuccessfully my js coding skillset

@ppigazzini This would need (a lot of) additonal code in `views.py` to work.

I swapped Later/Earlier as suggested by you

File: fishtest/fishtest/templates/nns.mak [hunk 1]
```diff
@@ -45,6 +50,15 @@ The recommended net for a given Stockfish executable can be found as the default
    <td style="text-align:right">${nn.get('downloads', 0)}</td>
   </tr>
   %endif
+  %endif
  %endfor
  </tbody>
 </table>
+<p>
```
Comment (ppigazzini) [inline fishtest/fishtest/templates/nns.mak:57]: This seems to work on DEV:
```
<p>
 %if prev_page:
  <a href="nns?page=${prev_page}">&lt Later networks</a>
 %endif
 %if prev_page and next_page:
  &#8209;
 %endif
 %if next_page:
  <a href="nns?page=${next_page}">Earlier networks &gt</a>
 %endif
</p>
```

Comment (tomtor) [reply fishtest/fishtest/templates/nns.mak:57]: @ppigazzini Looks much better! Thanks, will add that.

Comment (ppigazzini) [reply fishtest/fishtest/templates/nns.mak:57]: Please double check my syntax :)

Comment (tomtor) [reply fishtest/fishtest/templates/nns.mak:57]: I spotted the missing `;` but for some reason `&#8209;` shows also red in my FF page source view?

Comment (ppigazzini) [reply fishtest/fishtest/templates/nns.mak:57]: Perhaps `<span>&#8209;</span>` ?
https://www.toptal.com/designers/htmlarrows/punctuation/non-breaking-hyphen/

Comment (ppigazzini) [reply fishtest/fishtest/templates/nns.mak:57]: DEV updated with `<span>` tag.

Comment (tomtor) [reply fishtest/fishtest/templates/nns.mak:57]: `<span>`looks good

Comment (vondele) [reply fishtest/fishtest/templates/nns.mak:57]: maybe 'newer' and 'older' are better than 'earlier' and 'later' ...

**Comment by ppigazzini** [comment]
PROD updated, thank you @tomtor :)

**Comment by tomtor** [comment]
Thank you @ppigazzini :)

**Comment by SFisGOD** [comment]
@tomtor Thanks! I think it's better that the default view only shows the green nets?

**Comment by tomtor** [comment]
> @tomtor Thanks! I think it's better that the default view only shows the green nets?

@vondele prefered it this way, I think I do too. It remembers your choice, so it's not a real issue.

File: fishtest/fishtest/templates/nns.mak (hunk 1 above)
Comment (ppigazzini) [reply fishtest/fishtest/templates/nns.mak:57]: @vondele done.

image

**Pull request merged**